### PR TITLE
XP Globes - Move progress bar display from tooltip to globe

### PR DESCRIPTION
**Pull request by psikoi**
The old progress bar was annoying me apparently @deathbeam too, I decided to try and improve the progress display.

Started by refactoring slightly the logic on the "on hover" rendering as it made little sense.

I also had to move the rendering of the skill image as it was being displayed on top of the dark overlay, the whole point of the dark overlay was to darken the background so that the white text was very readable.

I'm sure there are going to be people that complain about loosing the decimal places, but the exp tracker also does not show them and I haven't heard of that being an issue yet, who cares if it's 25,6 or 25, come on.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java [hunk 1]
```diff
@@ -108,6 +110,22 @@ private void renderProgressCircle(Graphics2D graphics, XpGlobe skillToDraw, int
...
+		drawSkillImage(graphics, skillToDraw, x, y);
+
+		Point mouse = client.getMouseCanvasPosition();
+		int mouseX = mouse.getX() - bounds.x;
+		int mouseY = mouse.getY() - bounds.y;
+
+		// If mouse is hovering the globe
+		boolean hovering = backgroundCircle.contains(mouseX, mouseY);
+
+		if (hovering)
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: Couldn't this be moved into the other if statement below?

Comment (psikoi) [reply runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: Nah, different rendering orders

Comment (psikoi) [reply runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: Actually, fixed.

File: runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java [hunk 2]
```diff
@@ -108,7 +110,30 @@ private void renderProgressCircle(Graphics2D graphics, XpGlobe skillToDraw, int
...
 		Ellipse2D backgroundCircle = drawEllipse(graphics, x, y);
 
+		drawSkillImage(graphics, skillToDraw, x, y);
+
+		Point mouse = client.getMouseCanvasPosition();
+		int mouseX = mouse.getX() - bounds.x;
+		int mouseY = mouse.getY() - bounds.y;
+
 		Object renderHint = graphics.getRenderingHint(RenderingHints.KEY_STROKE_CONTROL);
+		graphics.setRenderingHint(RenderingHints.KEY_STROKE_CONTROL, renderHint);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: Hmm? What is this for

Comment (Kamielvf) [reply runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: This was here before so he probably would not know

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java]: `graphics.setRenderingHint(RenderingHints.KEY_STROKE_CONTROL, renderHint);` was definitely not here. It was after VALUE_STROKE_PURE where it makes sense, but here it doesnt

**Comment by deathbeam** [comment]
Any update @psikoi ?

**Comment by psikoi** [comment]
@deathbeam Yea fixed, removed those two lines.

**Pull request merged**